### PR TITLE
feat: support hierarchical imagePullSecrets configuration in AlluxioRuntime

### LINKED ISSUE #4221: [FEATURES] Support hierarchical imagePullSecrets configuration in AlluxioRuntime CRD
**What feature you'd like to add:**
We request the enhancement of the AlluxioRuntime to support hierarchical `imagePullSecrets` configuration. Specifically, we want to be able to specify `imagePullSecrets` at two levels:
1. Global level under `spec`
2. Component level
     a. Master component level under `spec.master`
     b. Worker component level under `spec.worker`
     c. FUSE component level under `spec.fuse`

```yaml
apiVersion: data.fluid.io/v1alpha1
kind: AlluxioRuntime
metadata:
  name: hbase
spec:
  replicas: 2
  imagePullSecrets: 
    - name: test-1
    - name: test-2
  tieredstore:
    levels:
      - mediumtype: MEM
        path: /dev/shm
        quota: 2Gi
        high: "0.95"
        low: "0.7"
  master:
    replicas: 3
    imagePullSecrets: 
    - name: test-1
    - name: test-2
  worker:
    imagePullSecrets: 
    - name: test-1
    - name: test-2
```

**Why is this feature needed:**
This feature will provide greater flexibility for users who need to manage different image pull secrets for various components of their AlluxioRuntime deployment. For example, users might have different private registries or credentials for the master and worker components. This hierarchical configuration will allow for more granular control and easier management of image pull secrets in complex deployment environments.

-

### PR DESCRIPTION
**Pull request by EvanCley**
### Ⅰ. Describe what this PR does
feature #4221
Support hierarchical imagePullSecrets configuration in AlluxioRuntime CRD

### Ⅱ. Does this pull request fix one issue?

fixes #4221

### Ⅲ. List the added test cases (unit test/integration test) if any, please explain if no tests are needed.
add test cases in TestGenerateAlluxioValueFil()/TestTransformMaster()/TestTransformWorker()/TestParseFuseImage()

### Ⅳ. Describe how to verify it
create several secrets and add below fields in runtime.yaml, then create the AlluxioRuntime and describe the field imagePullSecrets in master/worker/fuse pod
```
apiVersion: data.fluid.io/v1alpha1
kind: AlluxioRuntime
metadata:
  name: hbase
spec:
  replicas: 1
  tieredstore:
    levels:
      - mediumtype: MEM
        path: /dev/shm
        quota: 2Gi
        high: "0.95"
        low: "0.7"
  imagePullSecrets:
    - name: "secret-runtime"
  master:
    imagePullSecrets:
      - name: "secret-master"
  worker:
    imagePullSecrets:
      - name: "secret-worker"
  fuse:
    imagePullSecrets:
      - name: "secret-fuse" 
```

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Comment by cheyang** [comment]
@EvanCley CRD validation failed. Please use 'make update-crd' to keep CRDs latest

**Comment by EvanCley** [comment]
> @EvanCley CRD validation failed. Please use 'make update-crd' to keep CRDs latest

OK, have done

File: charts/alluxio/templates/fuse/daemonset.yaml [hunk 1]
```diff
@@ -64,9 +64,12 @@ spec:
         {{- end }}
         {{- end }}
     spec:
-      {{- with .Values.imagePullSecrets }}
+      {{- if .Values.master.imagePullSecrets}}
       imagePullSecrets:
-        {{- toYaml . | nindent 8 }}
+        {{- toYaml .Values.master.imagePullSecrets | nindent 8 }}
+      {{- else if .Values.imagePullSecrets}}
```
Comment (cheyang) [inline charts/alluxio/templates/fuse/daemonset.yaml:70]: Add a whitespace before }} in the template directive.

**Review by cheyang — CHANGES_REQUESTED** [review]

File: charts/alluxio/templates/fuse/daemonset.yaml (hunk 1 above)
Comment (xliuqq) [reply charts/alluxio/templates/fuse/daemonset.yaml:70]: use `fuse` not `master` imagePullSecrets

Comment (EvanCley) [reply charts/alluxio/templates/fuse/daemonset.yaml]: > use `fuse` not `master` imagePullSecrets

thanks for point that, have amended

Comment (EvanCley) [reply charts/alluxio/templates/fuse/daemonset.yaml]: > Add a whitespace before }} in the template directive.

ok, have done

**Comment by cheyang** [comment]
@allenhaozi Do you have any comments on this PR?

**Comment by allenhaozi** [comment]
> @allenhaozi Do you have any comments on this PR?

@cheyang LGTM

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**